examples/tv-app/android/java/ChannelManager.cpp

```diff
@@ -363,30 +371,273 @@ void ChannelManager::HandleGetProgramGuide(
+                recordFlag |= 4;
+            }
+
+            program->recordingFlag = MakeOptional(static_cast<uint32_t>(recordFlag));
+
+            needToFreePrograms.push_back(program);
+            tempPrograms.push_back(*program);
+        }
+
+        response.programList = DataModel::List<const ProgramType>(tempPrograms.data(), tempPrograms.size());
+
+        err = helper.Success(response);
+
+        for (ProgramType * program : needToFreePrograms)
+        {
+            delete program;
+        }
+
+        for (JniUtfString * str : needToFreeStrings)
+        {
+            delete str;
+        }
+    }
+
+exit:
+    ChipLogProgress(Zcl, " -- HandleGetProgramGuide -- 55");
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Zcl, "ChannelManager::HandleGetProgramGuide status error: %s", err.AsString());
+    }
 }
 
 bool ChannelManager::HandleRecordProgram(const chip::CharSpan & programIdentifier, bool shouldRecordSeries,
                                          const DataModel::DecodableList<AdditionalInfo> & externalIdList,
                                          const chip::ByteSpan & data)
 {
-    // Start recording
-    return true;
+    jboolean ret = JNI_FALSE;
+    JNIEnv * env = JniReferences::GetInstance().GetEnvForCurrentThread();
+
+    ChipLogProgress(Zcl, "Received ChannelManager::HandleRecordProgram");
+    VerifyOrExit(mChannelManagerObject != nullptr, ChipLogError(Zcl, "mChannelManagerObject null"));
+    VerifyOrExit(mRecordProgramMethod != nullptr, ChipLogError(Zcl, "mRecordProgramMethod null"));
+    VerifyOrExit(env != NULL, ChipLogError(Zcl, "env null"));
+
+    env->ExceptionClear();
+
+    {
+        UtfString jIdentifier(env, programIdentifier);
+
+        // NOTE: this example app does not pass the Data or ExternalIdList through to the Java layer
+        UtfString jData(env, "");
+        jobjectArray externalIDListArray = (jobjectArray) env->NewObjectArray(0, env->FindClass("java/util/Map$Entry"), NULL);
+
+        ret = env->CallBooleanMethod(mChannelManagerObject, mRecordProgramMethod, jIdentifier.jniValue(),
+                                     static_cast<jboolean>(shouldRecordSeries), externalIDListArray, jData.jniValue());
+        if (env->ExceptionCheck())
+        {
+            ChipLogError(DeviceLayer, "Java exception in ChannelManager::HandleRecordProgram");
+            env->ExceptionDescribe();
+            env->ExceptionClear();
+            return false;
+        }
+    }
+
+exit:
+    return static_cast<bool>(ret);
 }
 
 bool ChannelManager::HandleCancelRecordProgram(const chip::CharSpan & programIdentifier, bool shouldRecordSeries,
                                                const DataModel::DecodableList<AdditionalInfo> & externalIdList,
                                                const chip::ByteSpan & data)
 {
-    // Cancel recording
-    return true;
+    jboolean ret = JNI_FALSE;
+    JNIEnv * env = JniReferences::GetInstance().GetEnvForCurrentThread();
+
```

Comment:
Could you put the below JniLocalReferenceManager in the beginning of this HandleRecordProgram callback function so that we can automatically release local reference ,  when it is out of scope.
JNIEnv * env = JniReferences::GetInstance().GetEnvForCurrentThread();
VerifyOrReturn(env != nullptr, ChipLogError(Controller, "Could not get JNIEnv for current thread"));
JniLocalReferenceManager manager(env);

Reply:
fixed - thanks!